chall04/vde-dios.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+import sys
+import re
+
+
+def error(e , i):
+    file_name =  "stdin" if i == 0 else sys.argv[i]
+    if (e == 1):
+        print("%s: %s: Bad format" %(sys.argv[0], file_name))
+    elif (e == 2):
+        print("%s: %s: Can't read file" %(sys.argv[0], file_name))
+    elif (e == 3):
+        print("%s: %s: Not enough space in the given shelves" %(sys.argv[0], file_name))
+
+
+# Check format plus sort shelves
+def shelf_parse(shelves):
+    l = shelves.replace("\n", "").split(" ")
+    l.sort(key=len, reverse=True)
+    for c in l:
+        if not c.isdigit():
+            return None
+    return (l)
+
+
+# Check book format and add books
+def book_parse(books):
+    b = re.search('^\d+', books)
+    if re.search('^\d+\s\w+', books) is None:
+        return None
+    return int(b.group())
+
+
+# Calculate number of shelves: placing books on shells until completed
+def calc_number(shelves, books, i):
+    n_shelves = 0
+    capacity = 0
+    for c in shelves:
+        capacity += int(c)
+        n_shelves += 1
+        if capacity >= books:
+            break
+    if capacity < books:
+        error(3, i)
+    else:
+        print(n_shelves)
```

Comment:
I'm guessing you didn't understand the subject. You can't split books, and a book can't be on more than one shelf.  
Given this output your program will return `2` when it should return an error:
```
10 10
20 book
```